qiskit/pulse/channels.py

```diff
@@ -13,89 +13,107 @@
 # that they have been altered from the originals.
 
 """
-Channels.
+This module defines Pulse Channels. Channels include:
+
+  - transmit channels, which should subclass``PulseChannel``
+  - receive channels, such as ``AcquireChannel``
+  - non signal "channels" such as SnapshotChannel, MemorySlot and RegisterChannel, which relate
+    more closely to the classical control hardware.
+
+Novel channel types can often utilize the ``ControlChannel``, but if this is not sufficient, new
+channel types can be created. Then, they must be supported in the PulseQobj schema and the
+assembler.
 """
 from abc import ABCMeta
 
 from qiskit.pulse.exceptions import PulseError
 
 
 class Channel(metaclass=ABCMeta):
-    """Base class of channels."""
+    """Base class of channels. Channels provide a Qiskit-side label for typical quantum control
+    hardware signal channels. The final label -> physical channel mapping is the responsibility
+    of the hardware backend. For instance,``DriveChannel(0)`` holds instructions which the backend
+    should map to the signal line driving gate operations on the qubit labeled (indexed) 0.
+    """
 
-    prefix = None
+    prefix = None  # type: str
+    """A shorthand string prefix for characterizing the channel type."""
 
     def __init__(self, index: int):
         """Channel class.
 
         Args:
-            index: Index of channel
+            index: Index of channel.
 
         Raises:
-            PulseError: If index is not an integer
+            PulseError: If index is not a nonnegative integer.
         """
-        if not isinstance(index, int):
-            raise PulseError('Channel index must be integer')
-
+        if not isinstance(index, int) or index < 0:
+            raise PulseError('Channel index must be a nonnegative integer')
         self._index = index
 
     @property
     def index(self) -> int:
-        """Return the index of this channel."""
+        """Return the index of this channel. The index is a label for a control signal line
+        typically mapped trivially to a qubit index. For instance, ``DriveChannel(0)`` labels
```

Comment:
```suggestion
        typically mapped trivially to a qubit index. For instance, ``DriveChannel(0)`` creates a channel
```

Reply:
I'm describing the role of (0), not the instantiation process (I'm talking about physical hardware, which running code in Qiskit has no ability to create)